docpipeline/generate.py

```diff
@@ -101,8 +103,16 @@ def process_blob(blob, credentials, devsite_template):
     blob.download_to_filename(tar_filename)
     log.info(f"Downloaded gs://{blob.bucket.name}/{blob.name} to {tar_filename}")
 
-    tar.decompress(tar_filename, api_path)
-    log.info(f"Decompressed {blob.name} in {api_path}")
+    # Check to see if api directory exists in the tarball.
+    # If so, only decompress things into obj/*
+    tar_file = tarfile.open(tar_filename)
```

Comment:
I assume this is to allow a broader set of tars? (ones that are the contents of a directory vs a directory itself)

Is it possible we could have obj/api/api?

Is this only used by automation? Do we get diverse formats of tar through there?

Reply:
- yes
- without this snippet we could have `obj/api/api` if people were to include `api/*` in their tarballs, but we do not expect people to have `api/api/*` in the tarballs
- we don't get a diverse format, but we want to expand support where it makes sense, such as this case where the files already come packaged in `api/*` directory

Reply:
My concern is this expanded support would malform `api/api/*` tars right? since it would navigate inward? Are people manually creating tarsballs? I don't quite understand the motivation for expanding this if it is only used by bots. I may misunderstand how this is used.

Reply:
- It would, but the original would not have supported it either way and we do not expect people to have `api/api/*`at all.
- People are not manually creating tarballs afaik, but they're automating the process and curating how they are set up.
- Originally this support was included because all the tarballs we got just had everything laid flat out, but for `.net` Jon extracted things into `api/*` and `examples/*` and some laid flat out, which would not have been caught in the current format. So this support expands from just flat tarballs to ones that also have stuff laid in `api/*`

Reply:
👍